### PR TITLE
Add Sentry support for JavaScript

### LINKED ISSUE #6458: Add Sentry support
This needs further investigation.

## Context

The following alternatives were considered to implement Sentry support:
- **JS-side**, then error reports from Rust should be sent to JS code in order to upload them to Sentry. This raises some questions: how would Rust error reports be displayed? would they be sufficiently readable? 
- **Rust-side**, specific bindings should allow JS code to sent error reports to Rust in order to upload them to Sentry. Same questions apply as for the previous case.
- **Both JS & Rust** then two dedicated Sentry projects will run (one for JS, one for Rust) and error reports would be uploaded separately from JS and Rust code. 

The approach recommended by Sentry is to set up multiple projects for finer granularity:

> [How Many Projects Should I Create?](https://docs.sentry.io/organization/getting-started/#how-many-projects-should-i-create)
> 
> You could theoretically put all your errors into a single project, as this isn't limited in [sentry.io](https://sentry.io/)
> 
> However, setting up multiple projects to reflect your application with finer granularity helps makes errors more visible and actionable, which can have a big impact on your team's productivity.
> 
> Here are some points to consider:
> - If your application's source code is managed in multiple repositories, create a separate project for each repo.
> - If your app is made up of several micro-services, split them into projects accordingly.
> - If you have a monolithic codebase, set up separate projects for the backend and frontend.
> - Give each language its own project. For example, if your backend code contains NodeJS and Java components, separate those into two different projects.

## Initial POCs

Some tests were made in the following PRs: 
- Attempt to add Sentry JS-side: #5629
- Attempt to add Sentry Rust-side (bindings) #5566

## Current status

- [x] Create Sentry dedicated projects & teams
  - Project `parsec3-frontend` --notifies--> `scille-frontend`
  - Project `parsec3-libparsec` --notifies--> `scille-rust`
- [x] Setup Sentry in JavaScript code (Electron) & Rust bindings
- [ ] https://github.com/Scille/parsec-cloud/issues/7831
- [ ] Set Sentry DSN environment variables
- [ ] Fix Sentry issue with missing debug information

### PR DESCRIPTION
**Pull request by mmmarcos**
- [x] Setup Sentry in JavaScript code (Electron)
- [x] Set Sentry DSN environment variables

Closes #6458

### CODE REVIEW

File: bindings/electron/src/lib.rs [hunk 1]
```diff
@@ -10,6 +12,22 @@ lazy_static::lazy_static! {
         Mutex::new(tokio::runtime::Runtime::new().expect("Cannot start tokio runtime for libparsec"));
 }
 
+// TODO: Read env var?
+static SENTRY_LIBPARSEC_DSN: &str = "";
```
Comment (mmmarcos) [inline bindings/electron/src/lib.rs:16]: How can we read env var here?

File: client/electron/package.json [hunk 2]
```diff
@@ -22,11 +22,12 @@
     },
     "dependencies": {
         "@capacitor-community/electron": "^5.0.1",
+        "@sentry/electron": "^5.2.0",
         "chokidar": "~3.6.0",
         "electron-log": "^5.1.2",
         "electron-serve": "~1.3.0",
         "electron-unhandled": "~4.0.1",
-        "electron-updater": "~6.1.8",
+        "electron-updater": "6.1.8",
```
Comment (mmmarcos) [inline client/electron/package.json:30]: Not related to the PR but `electron-updater` was updated when adding `sentry/electron` and there seem to be a breaking change with `6.1.9`

File: client/electron/src/index.ts [hunk 3]
```diff
@@ -15,6 +15,8 @@ import { ElectronCapacitorApp, setupContentSecurityPolicy, setupReloadWatcher }
 
 const PARSEC_CONFIG_DIR_NAME = 'parsec3';
 const ELECTRON_CONFIG_DIR_NAME = electronIsDev ? 'app-dev' : 'app';
+// TODO: Read env var
+const SENTRY_ELECTRON_DSN = ''
```
Comment (mmmarcos) [inline client/electron/src/index.ts:19]: @Max-7 how to read env var here?

File: client/electron/src/setup.ts [hunk 4]
```diff
@@ -328,6 +331,26 @@ export class ElectronCapacitorApp {
...
+    //   // Set tracesSampleRate to 1.0 to capture 100%
+    //   // of transactions for performance monitoring.
+    //   // We recommend adjusting this value in production
+    //   tracesSampleRate: 1.0,
+
+    //   // Capture Replay for 10% of all sessions,
+    //   // plus for 100% of sessions with an error
+    //   replaysSessionSampleRate: 0.1,
+    //   replaysOnErrorSampleRate: 1.0,
+    // });
```
Comment (mmmarcos) [inline client/electron/src/setup.ts:352]: @Max-7 can you check where should we add this code?

File: client/electron/src/index.ts [hunk 5]
```diff
@@ -15,10 +15,31 @@ import { ElectronCapacitorApp, setupContentSecurityPolicy, setupReloadWatcher }
...
+function initSentry(): void {
+  // Sentry is initialized in
+  // - dev mode: only if the env var is set
+  // - release mode: always, either with the default DSN or with the env var if set
+  let sentry_dsn = '';
+  if (SENTRY_DSN_GUI_ELECTRON) {
+    sentry_dsn = SENTRY_DSN_GUI_ELECTRON;
+  } else if (!electronIsDev) {
+    sentry_dsn = SENTRY_DSN_GUI_ELECTRON_DEFAULT;
+  }
```
Comment (FirelightFlagboy) [inline client/electron/src/index.ts:33]:
```suggestion
  let sentry_dsn = SENTRY_DSN_GUI_ELECTRON;
  if (!sentry_dsn && !electronIsDev) {
    sentry_dsn = SENTRY_DSN_GUI_ELECTRON_DEFAULT;
  }
```

File: client/src/services/sentry.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  });
+}
+
+function disable(): void {
+  const client = SentryAPI.getClient();
+  if (client) {
+    enabled = false;
+    console.log('Disabling Sentry');
+  } else {
+    console.log('Sentry is not initialized, cant disable');
```
Comment (mmmarcos) [inline client/src/services/sentry.ts:43]:
```suggestion
    console.log('Sentry disabled');
  } else {
    console.log('Sentry is not initialized, can't disable');
```

File: client/src/services/sentry.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  }
+}
+
+function enable(): void {
+  const client = SentryAPI.getClient();
+  if (client) {
+    enabled = true;
+    console.log('Enabling Sentry');
+  } else {
+    console.log('Sentry is not initialized, cant enable');
```
Comment (mmmarcos) [inline client/src/services/sentry.ts:53]:
```suggestion
    console.log('Sentry enabled');
  } else {
    console.log('Sentry is not initialized, can't enable');
```

File: client/vite.config.ts [hunk 8]
```diff
@@ -49,6 +50,17 @@ if (platform === 'web') {
   plugins.push(wasmPack([{ path: '../bindings/web/', name: 'libparsec_bindings_web' }]));
 }
 
+if (process.env.VITE_SENTRY_AUTH_TOKEN) {
+  const sentryPlugin = sentryVitePlugin({
+    org: 'scille',
+    project: 'parsec3-frontend',
+    authToken: process.env.VITE_SENTRY_AUTH_TOKEN,
```
Comment (mmmarcos) [inline client/vite.config.ts:57]:
```suggestion
if (process.env.SENTRY_AUTH_TOKEN) {
  const sentryPlugin = sentryVitePlugin({
    org: 'scille',
    project: 'parsec3-frontend',
    authToken: process.env.SENTRY_AUTH_TOKEN,
```

Comment (Max-7) [reply client/vite.config.ts]: Vite blocks any env variable that is not prefixed by `VITE_` (to avoid exposing variables to the app)

**Review by Ironicbay — APPROVED** [review]

**Review by NicoTuxx — APPROVED** [review]

**Pull request merged**